[15.0] connector_importer: misc fix/imp

simahawk opened the pull request:
See atomic commits for details

simahawk committed:
c_importer: abandon need for unique_key_is_xmlid attribute

simahawk committed:
c_importer: sanitize xids w/ xmlid_to_rel

Make sure XIDs are always complete.
Make module customizable anyway.

simahawk committed:
c_importer: dynamic mapper handle selection field

simahawk committed:
c_importer: dynamic mapper get required defaults

Automatically load defaults for required keys that are missing.
You can still bypass this behavior by using 'source_key_empty_skip' option.

simahawk committed:
connector_importer: update ROADMAP

simahawk committed:
connector_importer: assume 'id' is always an xmlid

simahawk committed:
connector_importer: importer.record is the default one

simahawk committed:
connector_importer: update ROADMAP

sebalix commented on connector_importer/components/importer.py:
```diff
@@ -63,9 +63,8 @@ class RecordImporter(Component):
-    # if this key is an external/XML ID, set 'odoo_unique_key_is_xmlid' to True
+    # if this key is an external/XML ID, prefix the name with `xid::` (eg: xid::id)
     odoo_unique_key = ""
-    odoo_unique_key_is_xmlid = False
```
Should we add a deprecated warning or an error if we set this class attribute? Not sure to remember to change this when migrating custom code.

simahawk replied:
It cannot be just deprecated... I don't want to maintain it anymore.
We could raise an error on component init if that attribute is set.
It will break on your face explicitly if you are using it in custom code.

simahawk replied:
@sebalix fine for you with last commit? https://github.com/OCA/connector-interfaces/pull/98/commits/bed97993b3b82583b994c739f843b4d5959591b5

sebalix replied:
@simahawk all good, thanks!

sebalix commented on connector_importer/components/odoorecord.py:
```diff
@@ -14,7 +16,6 @@ class OdooRecordHandler(Component):
 
     # TODO: collect these from `work.options.record_handler`
     unique_key = ""
-    unique_key_is_xmlid = False
```
Same?

sebalix reviewed (approved):
LG, approving, one question however

simahawk committed:
connector_importer: break if unique_key_is_xmlid found

In the v15 refactoring we eliminated the need for 'unique_key_is_xmlid' attribute.
To ease the migration of existing modules, break immediately and let know people they have to fix it.

simahawk commented:
/ocabot merge minor
